### PR TITLE
Refactor checklists

### PR DESCRIPTION
**Pull request by xuanxu**
New command `@whedon generate my checklist` for reviewers

Related changes: openjournals/joss#996

### CODE REVIEW

File: whedon_api.rb [hunk 1]
```diff
@@ -390,6 +394,25 @@ def assign_version(version_string)
...
+    end
+
+    # update comment
+    submitting_author = issue.body.match(/\*\*Submitting author:\*\*\s*.(@\S*)/)[1]
+    repository_url = issue.body.match(/\*\*Repository:\*\*\s*<a href="([^"]*)"/)[1]
+    checklist = erb :reviewer_checklist, locals: { reviewer: @sender, submitting_author: submitting_author, repository_url: repository_url }
+    github_client.update_comment(@nwo, @comment_id, checklist)
+
+    # link checklist from issue's body
+    text = "[Checklist for @#{sender}](https://github.com/#{@nwo}/issues/#{@issue_id}#issuecomment-#{@comment_id})"
```
Comment (arfon) [inline whedon_api.rb:411]:
```suggestion
    text = "[Checklist for @#{@sender}](https://github.com/#{@nwo}/issues/#{@issue_id}#issuecomment-#{@comment_id})"
```

File: whedon_api.rb [hunk 2]
```diff
@@ -390,6 +394,25 @@ def assign_version(version_string)
...
+
+    # update comment
+    submitting_author = issue.body.match(/\*\*Submitting author:\*\*\s*.(@\S*)/)[1]
+    repository_url = issue.body.match(/\*\*Repository:\*\*\s*<a href="([^"]*)"/)[1]
+    checklist = erb :reviewer_checklist, locals: { reviewer: @sender, submitting_author: submitting_author, repository_url: repository_url }
+    github_client.update_comment(@nwo, @comment_id, checklist)
+
+    # link checklist from issue's body
+    text = "[Checklist for @#{@sender}](https://github.com/#{@nwo}/issues/#{@issue_id}#issuecomment-#{@comment_id})"
+    new_body = issue_body + "\n" + text
```
Comment (arfon) [inline whedon_api.rb:412]:
```suggestion
    new_body = issue.body + "\n" + text
```

**Review by xuanxu — COMMENTED** [review]
Specs ready! 
Build passing in Ruby 2.6.6 and 2.7.4

**Review by xuanxu — COMMENTED** [review]
This repo is deprecated now in favor of https://github.com/openjournals/buffy.
Merging changes before archiving

**Pull request merged**